src/Controller/UserController.php

```diff
@@ -41,9 +47,31 @@ public function addUsers(CreateUser $createuser): Response
             $this->entityManager->persist($user);
             $this->entityManager->flush();
         } catch (\Exception $ex) {
-            return $this->json($ex->errorInfo, 404);
+            return $this->json($ex->errorInfo, Response::HTTP_BAD_REQUEST);
```

Comment:
instead of return json response, you could just throw BadRequestExecption with proper message.

Reply:
fixed